rickchurch/church.py

```diff
@@ -105,22 +110,37 @@ async def authorize() -> fastapi.Response:
 @app.get("/oauth_callback", include_in_schema=False)
 async def auth_callback(request: fastapi.Request) -> fastapi.Response:
     """This endpoint is only used as a redirect target from discord OAuth2."""
+    client = request.app.state.client
     code = request.query_params["code"]
     try:
-        user = await get_oauth_user(code)
+        user, access_token = await get_oauth_user(client, code)
         token = await add_user(user, request.state.db_conn)
     except PermissionError:
         # `add_user` can return `PermissionError` if the user already has a token, which is banned.
         raise fastapi.HTTPException(401, "You are banned")
 
+    # Join them into the rickchurch server
+    res = await client.put(
+        constants.DISCORD_BASE_URL + f"/guilds/{constants.discord_guild_id}" + f"/members/{user['id']}",
+        json={"access_token": access_token},
+        headers={"Authorization": f"Bot {constants.discord_bot_token}"},
+    )
+    if res.status_code not in [200, 201, 204]:
+        try:
+            text = res.text
+        except Exception:
+            logger.exception("Failed getting text of response")
+            text = "<error>"
+        logger.error(f"Joining server for user failed: Code {res.status_code} text {text!r}")
```

Comment:
You shouldn't except bare `Exception`, be specific. That way, it will be clear what is going on here, this looks like it should be `AttributeError`.

Also, the `text = "<error>"` is a bit weird, I'd prefer `text = "N/A"` or something similar, so it's immediately obvious that the text isn't available.

Reply:
It is meant to catch text decoding errors but I'm not sure which exception that would be but not `AttributeError`, I'd have to do some research. Can you elaborate on the 3rd point, I don't quite follow.

Reply:
`UnicodeDecodeError`?

Reply:
Well if we are unable to decode the text, instead of making an exception log, we can just set `text="N/A"` to signify that it is for some reason unavailable and log error normally, no need to log the exception together with the error afterwards